File: problems/Longest Continuous Increasing Subsequence/findLengthOfLCIS.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLengthOfLCIS = function(nums) {
    if (!nums || !nums.length) {
        return 0;
    }

    let result = 0;
    let anchor = 0;

    for (let i = 0; i in nums; ++i) {
        if (i && nums[i - 1] >= nums[i]) {
            anchor = i;
        }
        
        result = Math.max(result, i - anchor + 1);
    }

    return result;
};

File: problems/Longest Continuous Increasing Subsequence/findLengthOfLCIS.java
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        if (null == nums || 0 == nums.length) {
            return 0;
        }

        int result = 1;
        int anchor = 0;

        for (int idx = 1; nums.length > idx; ++idx) {
            if (nums[idx - 1] >= nums[idx]) {
                anchor = idx;
            }

            result = Math.max(result, 1 + idx - anchor);
        }

        return result;
    }
}
